caffe2/operators/spatial_batch_norm_op.h

```diff
@@ -52,7 +52,7 @@ class SpatialBNOp : public Operator<Context> {
     const auto& bias = Input(BIAS);
 
     const int ndim = X.dim();
-    CAFFE_ENFORCE_GE(ndim, 3);
+    CAFFE_ENFORCE_GE(ndim, 2);
```

Comment:
I am not sure I understand why this needs to be changed here. Could you explain what was the issue with the Caffe2 op and how this helps.

Reply:
Based on PyTorch spec, Batch norm 1D can have a shape of (N, C) or (N, C, L). 
Since opset 9, ONNX supports (N, C) shape. However, looks like Caffe2 and the symbolic function have not been updated accordingly.
That's why now we are able to remove the branching for rank==2 in the symbolic,
and we are able to update Caffe 2 with this change.
I'm going to add more tests to cover all cases for ORT and Caffe2.
I'll make sure both set of tests pass.
@houseroad in case this breaks any internal tests for Caffe2, please let us know and we'll revert this.

Reply:
cc @spandantiwari
Please review the added tests

Reply:
Sounds good.